### PR TITLE
Mushroom prefab

### PR DESCRIPTION
**Pull request by gleblebedev**
Mushroom prefab instead of setting up a mushroom from code.
Default skybox texture and material.

### CODE REVIEW

**Comment by rokups** [comment]
I do not think this is a good idea. Samples are there to show how something is done. They are supposed to be simple yet explicit. To showcase prefab use there should be a new sample that does just that.

File: Source/Samples/04_StaticScene/StaticScene.cpp [hunk 1]
```diff
@@ -99,15 +100,14 @@ void StaticScene::CreateScene()
...
     for (unsigned i = 0; i < NUM_OBJECTS; ++i)
     {
         Node* mushroomNode = scene_->CreateChild("Mushroom");
         mushroomNode->SetPosition(Vector3(Random(90.0f) - 45.0f, 0.0f, Random(90.0f) - 45.0f));
         mushroomNode->SetRotation(Quaternion(0.0f, Random(360.0f), 0.0f));
         mushroomNode->SetScale(0.5f + Random(2.0f));
-        auto* mushroomObject = mushroomNode->CreateComponent<StaticModel>();
-        mushroomObject->SetModel(cache->GetResource<Model>("Models/Mushroom.mdl"));
-        mushroomObject->SetMaterial(cache->GetResource<Material>("Materials/Mushroom.xml"));
+        mushroomNode->CreateComponent<PrefabReference>()->SetPrefab(mushroomPrefab);
```
Comment (eugeneko) [inline Source/Samples/04_StaticScene/StaticScene.cpp:110]: Please don't do double -> in samples

**Comment by gleblebedev** [comment]
> > I do not think this is a good idea. Samples are there to show how something is done. They are supposed to be simple yet explicit. To showcase prefab use there should be a new sample that does just that.
> 
> I somewhat agree. Samples are only marginally shorter (2 lines instead of 3). I suggest to use prefabs only where samples use physics part of the prefab.

The point of the prefab - to have something reusable. Is setting up one resource instead of two demonstrates convenience of the approach.

**Comment by gleblebedev** [comment]
> I do not think this is a good idea. Samples are there to show how something is done. They are supposed to be simple yet explicit. To showcase prefab use there should be a new sample that does just that.

I can extrapolate the argument and say that we shouldn't use light source in Character Demo because we have a dedicated sample for the light sources.

Where is the line that separates appropriate use for inappropriate?

**Comment by rokups** [comment]
Light source in character demo is to make sample not look like crap. You can say same about boxes in the sample. They arent necessary, but they are a useful filler. Using prefabs for irrelevant filler may be OK. However places like [this](https://github.com/rbfx/rbfx/pull/423/files#diff-6f851212eade78840733d12538dfd0453ea351e03abf261a800a818f87cfd76cL150-L158) teach user how to do multiple useful things. Hiding all that in a prefab is a good idea in any project except for samples.

**Comment by gleblebedev** [comment]
> Light source in character demo is to make sample not look like crap. You can say same about boxes in the sample. They arent necessary, but they are a useful filler. Using prefabs for irrelevant filler may be OK. However places like [this](https://github.com/rbfx/rbfx/pull/423/files#diff-6f851212eade78840733d12538dfd0453ea351e03abf261a800a818f87cfd76cL150-L158) teach user how to do multiple useful things. Hiding all that in a prefab is a good idea in any project except for samples.

So... Should I roll it back too and make a dedicated prefab sample, is this what you suggest?

**Comment by gleblebedev** [comment]
Also a contradicting thing is a platformer sample. It doesn't showcase just one aspect of an engine but shows how several engine features work together to make something that resembles a game. That's why I've added prefabs to character demo as it shows how a game code would look like.

**Comment by gleblebedev** [comment]
> That is a fair point. If overwritten stuff is showcased in other samples i wont argue against the changes. Cant check right now 😁

Is it not showcased in the same file for cubes?

**Pull request merged**